### PR TITLE
paperless-ngx: 1.13.0 -> 1.14.0

### PR DESCRIPTION
**Pull request by e1mo**
###### Description of changes

Most notably: Multi user permission support. Plethora of various small to medium size changes to various parts:
https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.14.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/office/paperless-ngx/default.nix [hunk 1]
```diff
@@ -17,19 +17,27 @@
...
-      django = super.django_4;
+      django = super.django_4.overridePythonAttrs (oldAttrs: rec {
+        version = "4.1.8";
+        src = oldAttrs.src.override {
+          inherit version;
+          hash = "sha256-QyU/S0osOhaUzVLTUXhHJxk0yE29Bgyg6CqfkOVp2rM=";
+        };
+
+        doCheck = false;
+      });
```
Comment (mweinelt) [inline pkgs/applications/office/paperless-ngx/default.nix:40]: Any problems with Django 4.2?

File: pkgs/applications/office/paperless-ngx/default.nix [hunk 2]
```diff
@@ -74,6 +70,18 @@ let
...
+      # upstreams pipfile.lock uses the 0.3.0 version, but with that, the build / test phase fails in NixOS
+      djangorestframework-guardian = super.djangorestframework-guardian.overridePythonAttrs (oa: rec {
+        version = "0.5.0";
+        src = fetchFromGitHub {
+          owner = "johnthagen";
+          repo = "django-rest-framework-guardian2";
+          rev = "v${version}";
+          hash = "sha256-aW20xEmVTAgwayWMJsabmyKNW65NftJyQANtT6JV74U=";
+        };
+      });
```
Comment (mweinelt) [inline pkgs/applications/office/paperless-ngx/default.nix:84]: Properly package that up as `djangorestframework-guardian2`.

https://github.com/johnthagen/django-rest-framework-guardian2/blob/master/setup.cfg#L24

File: pkgs/applications/office/paperless-ngx/default.nix [hunk 3]
```diff
@@ -240,7 +251,8 @@ python.pkgs.buildPythonApplication rec {
   ]
   ++ redis.optional-dependencies.hiredis
   ++ twisted.optional-dependencies.tls
-  ++ uvicorn.optional-dependencies.standard;
+  ++ uvicorn.optional-dependencies.standard
+  ++ lib.optional stdenv.hostPlatform.isx86 zxing_cpp; # Upstream only has this for x86 systems
```
Comment (mweinelt) [inline pkgs/applications/office/paperless-ngx/default.nix:255]: Builds fine on aarch64-linux. What is the reason upstream has this only for x86?

File: pkgs/applications/office/paperless-ngx/default.nix (hunk 1 above)
Comment (e1mo) [reply pkgs/applications/office/paperless-ngx/default.nix:40]: Paperless-ngx itself explicitly uses `django~=4.1`: 
https://github.com/paperless-ngx/paperless-ngx/blob/405769dc976942305533d766afc6a0a872b6cf45/Pipfile#L13

I've just ran it with django 4.2, that seems to have worked out (both building and tests). So I guess we can run it with 4.2 too.

File: pkgs/applications/office/paperless-ngx/default.nix (hunk 3 above)
Comment (e1mo) [reply pkgs/applications/office/paperless-ngx/default.nix:255]: [The docs](https://docs.paperless-ngx.com/configuration/) only state 

> `PAPERLESS_CONSUMER_BARCODE_SCANNER=<string>`
> [...]
> zxing is not available on all platforms

But when you have tested it (I sadly don't have any arm aarch hosts yet, so I wasn't able to test it) we can also include it for every system.

Comment (mweinelt) [reply pkgs/applications/office/paperless-ngx/default.nix:255]: Let's include it unconditionally then.

Comment (gador) [reply pkgs/applications/office/paperless-ngx/default.nix]: For reference: This is why `zxing` isn't included for all architectures: https://github.com/paperless-ngx/paperless-ngx/pull/2903#issuecomment-1472714824

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 228011` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>5 packages failed to build and are new build failures:</summary>
  <ul>
<li>paperless-ngx: log was empty</li>
<li>python310Packages.djangorestframework-guardian2: log was empty</li>
<li>python310Packages.djangorestframework-guardian2.dist: log was empty</li>
<li>python311Packages.djangorestframework-guardian2: log was empty</li>
<li>python311Packages.djangorestframework-guardian2.dist: log was empty</li>
  </ul>
</details>

**Comment by gador** [comment]
This is the error leading up to the failed build:
```
these 6 derivations will be built:
  /nix/store/9sqrlv42q1s2dqp43d4ihza7al33q6mj-source.drv
  /nix/store/2wm8764w83mgp0pjxm1rjq3nlpd8x4hj-python3.11-djangorestframework-guardian2-0.5.0.drv
  /nix/store/3cvgbpzk5qj8j27k7w6lxnfjkh965clw-python3.10-djangorestframework-guardian2-0.5.0.drv
  /nix/store/6xr2wkilz7yxqbrp79iwjsqhl7x31nn5-python3.10-djangorestframework-guardian2-0.5.0.drv
  /nix/store/p42m37lylipf92q2h4l1acvxbj5bdra5-paperless-ngx-1.14.0.drv
  /nix/store/c47vw3sgl755i2za54a5caq2xw1swnc8-review-shell.drv
resolved derivation: '/nix/store/9sqrlv42q1s2dqp43d4ihza7al33q6mj-source.drv' -> '/nix/store/vlfj5ngm074z0nrs983qy0ihkbpsmlsm-source.drv'
source> building '/nix/store/vlfj5ngm074z0nrs983qy0ihkbpsmlsm-source.drv'
source>
source> trying https://github.com/johnthagen/django-rest-framework-guardian2/archive/0.5.0.tar.gz
source>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
source>                                  Dload  Upload   Total   Spent    Left  Speed
source>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
source>   0    14    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
source> curl: (22) The requested URL returned error: 404
source> error: cannot download source from any mirror
error: builder for '/nix/store/vlfj5ngm074z0nrs983qy0ihkbpsmlsm-source.drv' failed with exit code 1;
       last 8 log lines:
       >
       > trying https://github.com/johnthagen/django-rest-framework-guardian2/archive/0.5.0.tar.gz
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       >   0    14    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (22) The requested URL returned error: 404
       > error: cannot download source from any mirror
       For full logs, run 'nix log /nix/store/vlfj5ngm074z0nrs983qy0ihkbpsmlsm-source.drv'.
error: 1 dependencies of derivation '/nix/store/3cvgbpzk5qj8j27k7w6lxnfjkh965clw-python3.10-djangorestframework-guardian2-0.5.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6xr2wkilz7yxqbrp79iwjsqhl7x31nn5-python3.10-djangorestframework-guardian2-0.5.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2wm8764w83mgp0pjxm1rjq3nlpd8x4hj-python3.11-djangorestframework-guardian2-0.5.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/p42m37lylipf92q2h4l1acvxbj5bdra5-paperless-ngx-1.14.0.drv' failed to build
error: 3 dependencies of derivation '/nix/store/c47vw3sgl755i2za54a5caq2xw1swnc8-review-shell.drv' failed to build
```

**Comment by e1mo** [comment]
Thanks @gador ! Copy and paste oversights… It's fixed now and `nix build .#python3Packages.djangorestframework-guardian2.src --rebuild` works as intended.

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 228011` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>paperless-ngx</li>
<li>python310Packages.djangorestframework-guardian2</li>
<li>python310Packages.djangorestframework-guardian2.dist</li>
<li>python311Packages.djangorestframework-guardian2</li>
<li>python311Packages.djangorestframework-guardian2.dist</li>
</ul>
</details>

**Pull request merged**